### DISCUSSION TITLE
Automatic signing

### DISCUSSION THREAD

**Discussion by varunsh-coder** [step-security/harden-runner#77] [Ideas]
Would you like `harden-runner` to automatically sign your build artifact? [Signing releases](https://github.com/ossf/scorecard/blob/main/docs/checks.md#signed-releases) is a Scorecards requirement, but is hard to do. 

@jauderho has created a sample workflow for key-less signing and SBOM generation here: 
https://github.com/jauderho/dockerfiles/blob/main/.github/workflows/age.yml
https://github.com/jauderho/dockerfiles/actions/runs/1755633128

**Comment by jauderho** [comment]
So I've noticed that the cosign process is creating additional entries on the registries causing my shields.io badge to not display properly. 

It's unclear to me right now if it is something I'm doing incorrectly, the shields team will have to filter out or new functionality that registries will have to support/implement.

More information captured here: https://github.com/docker/roadmap/issues/269#issuecomment-1024700813

**Comment by jauderho** [comment]
Now tracking my findings here: https://github.com/jauderho/dockerfiles/discussions/149
